05.wc/wc.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  paths, target_options = options_and_paths
+  text_metadata_collection =
+    if paths.empty?
+      [create_text_metadata(input: $stdin.read)]
+    else
+      build_text_metadata(paths)
+    end
+  width = calculate_output_width(text_metadata_collection, target_options)
+  text_metadata_collection.each do |text_metadata|
+    render(target_options, text_metadata, width)
+  end
+end
+
+def options_and_paths
+  line = false
+  word = false
+  size = false
+  opt = OptionParser.new
+  opt.on('-l') { |v| line = v }
+  opt.on('-w') { |v| word = v }
+  opt.on('-c') { |v| size = v }
+  paths = opt.parse(ARGV)
+  target_options =
+    if [line, word, size].none?
+      %i[line_count word_count size]
+    else
+      { line_count: line, word_count: word, size: size }.map do |option, flag|
+        option if flag
+      end.compact
+    end
+  [paths, target_options]
+end
+
+def create_text_metadata(input:, name: nil)
```

Comment:
```suggestion
def create_text_metadata(input, name = nil)
```

キーワード引数じゃなくてもいいんじゃないかなー、と思います
https://bootcamp.fjord.jp/pages/dont-abuse-keyword-args

Reply:
[キーワード引数である必要がなくなったので修正](https://github.com/s-tone-gs/ruby-practices/pull/10/commits/fa413caa8919af177ff1cbed8b075c2e561c8b21)にて反映しました！